modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisFilterToSQL.java

```diff
@@ -92,6 +94,21 @@ protected void visitLiteralGeometry(Literal expression) throws IOException {
     }
 
     @Override
+    protected Object evaluateLiteral(Literal expression, Class target) {
+        Date converted = null;
+        if (target == PostGISDialect.XDate.class) {
+            converted = Converters.convert(expression.evaluate(null), Date.class);
```

Comment:
And over here the "bigdata" placeholder is being used as a bit of a trick (violating the API contract of evaulateLiteral since a XDate is not actually being returned.